packages/codemirror/src/editor.ts

```diff
@@ -252,7 +252,21 @@ export class CodeMirrorEditor implements CodeEditor.IEditor {
     );
 
     // every time the model is switched, we need to re-initialize the editor binding
-    this.model.sharedModelSwitched.connect(this._initializeEditorBinding, this);
+    this.model.sharedModelSwitched.connect(() => {
+      this._initializeEditorBinding();
+      Private.updateEditor(
+        this._editor,
+        fullConfig,
+        this._yeditorBinding,
+        this._editorConfig,
+        [
+          this._markField,
+          Prec.high(domEventHandlers),
+          updateListener,
+          translation
+        ]
```

Comment:
These arguments are identical to the arguments for `Private.createEditor` See L242~L251